File: app/src/main/java/com/juliuscanute/multiconfig/ui/config/ConfigurationState.kt
package com.juliuscanute.multiconfig.ui.config

import com.juliuscanute.multiconfig.ui.adapter.ConfigurationViewDataModel

sealed class ConfigurationState {
    data class SelectedConfigurationState(val environment: String) : ConfigurationState()
    data class LoadApplicationConfigurationState(val items: List<ConfigurationViewDataModel>, val environment: String) :
        ConfigurationState()
}

File: iosApp/iosApp/Configuration/Views/ConfigurationState.swift
//
//  ConfigurationState.swift
//  iosApp
//
//  Created by Julius Canute on 29/3/20.
//

import Foundation

enum ConfigurationState {
    case initialState
    case appConfig(LoadApplicationConfigurationState)
    case selectedConfig(String)
    case buttonConfig(ButtonConfigurationState)
}

struct ButtonConfigurationState {
    let environment: String
}

struct LoadApplicationConfigurationState {
    let items: [ConfigurationViewDataModel]
    let environment: String
    let selectedIndex: Int
}

struct ConfigurationViewDataModel {
    let index: Int
    let environment: String
    let selected: Bool
}
